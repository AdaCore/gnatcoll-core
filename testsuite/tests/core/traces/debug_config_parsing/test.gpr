with "gnatcoll";

project Test is
   for Object_Dir use "obj";
   for Exec_Dir use ".";
   for Main use ("test.adb");
end Test;
