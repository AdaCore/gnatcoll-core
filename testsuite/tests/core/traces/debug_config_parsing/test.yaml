driver: build_run_diff
description: Check behavior of DEBUG configuration variants
control:
    - [SKIP, "env.is_cross", "Tests using test.sh currently not supported on cross targets, see T616-039"]
