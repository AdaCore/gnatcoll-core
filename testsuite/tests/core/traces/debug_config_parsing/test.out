== DEBUG.LOCATION=yes
[PKG] Entering Test.Foo:test.adb:13 (loc:test.adb:13)
   [PKG] A = 3 (loc:test.adb:15)
   [PKG] Entering Test.Foo:test.adb:13 (loc:test.adb:13)
      [PKG] A = 2 (loc:test.adb:15)
      [PKG] Entering Test.Foo:test.adb:13 (loc:test.adb:13)
         [PKG] A = 1 (loc:test.adb:15)
      [PKG] Leaving Test.Foo:test.adb:13 (loc:test.adb:13)
   [PKG] Leaving Test.Foo:test.adb:13 (loc:test.adb:13)
[PKG] Leaving Test.Foo:test.adb:13 (loc:test.adb:13)
[PKG] [Test success msg] (loc:test.adb:25)
[PKG.EXCEPTIONS] [Test error msg] (loc:test.adb:33)

== DEBUG.ENCLOSING_ENTITY=yes
[PKG] Entering Test.Foo:test.adb:13 (entity:Test.Foo)
   [PKG] A = 3 (entity:Test.Foo)
   [PKG] Entering Test.Foo:test.adb:13 (entity:Test.Foo)
      [PKG] A = 2 (entity:Test.Foo)
      [PKG] Entering Test.Foo:test.adb:13 (entity:Test.Foo)
         [PKG] A = 1 (entity:Test.Foo)
      [PKG] Leaving Test.Foo:test.adb:13 (entity:Test.Foo)
   [PKG] Leaving Test.Foo:test.adb:13 (entity:Test.Foo)
[PKG] Leaving Test.Foo:test.adb:13 (entity:Test.Foo)
[PKG] [Test success msg] (entity:Test)
[PKG.EXCEPTIONS] [Test error msg] (entity:Test)

== DEBUG.COLORS=yes
[PKG] Entering Test.Foo:test.adb:13
   [PKG] A = 3
   [PKG] Entering Test.Foo:test.adb:13
      [PKG] A = 2
      [PKG] Entering Test.Foo:test.adb:13
         [PKG] A = 1
      [PKG] Leaving Test.Foo:test.adb:13
   [PKG] Leaving Test.Foo:test.adb:13
[PKG] Leaving Test.Foo:test.adb:13
[PKG] [Test success msg]
[PKG.EXCEPTIONS] [Test error msg]
